hedera-node/test-clients/src/main/java/com/hedera/services/bdd/junit/hedera/subprocess/SubProcessNetwork.java

```diff
@@ -545,4 +765,18 @@ private Map<Long, Long> latestCandidateWeights() {
             throw new IllegalStateException(e);
         }
     }
+
+    private int findAvailablePort() {
+        // Find a random available port between 30000 and 40000
+        int attempts = 0;
+        while (attempts < 100) {
+            int port = RANDOM.nextInt(FIRST_CANDIDATE_PORT, LAST_CANDIDATE_PORT);
+            try (ServerSocket socket = new ServerSocket(port)) {
+                return port;
```

Comment:
I guess this works...but isn't it pretty expensive to open a Socket, just to see if the port is free?